types/init.lua

```diff
@@ -368,7 +368,14 @@ local FontAttributes = {
 ---@field state "Charging" | "Discharging" | "Empty" | "Full" | "Unknown"
 
 ---@class WeztermPlugin
----@field require fun(url: string): any
+---@field require fun(url: string): any   -- 'require' function: takes a URL (string) and returns some result (any type)
```

Comment:
While this is correct, convention dictates that the module should return a `apply_to_config` function that accepts at least a config builder parameter, but may
pass other parameters, or a lua table with a `config` field that maps
to a config build parameter.

So I guess this would be more accurate:

```suggestion
---@class PluginResponse
---@field apply_to_config fun(config: Config, ...: any): any Function that accepts at least a config builder parameter, but may pass other parameters, or a lua table with a `config` field that maps to a config build parameter.

---@field require fun(url: string): PluginResponse Takes a plugin repo URL (string). This plugin has to return a `apply_to_config` function that accepts at least a config builder parameter
```

I'm basing all this on [this commit](https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd)